### PR TITLE
Fix package

### LINKED ISSUE #12: Package breaks on DelaunayTriangulation 1.0, and question about code
The package currently breaks since on DelaunayTriangulation 1.0 `indices` has now been deleted. I'll get a PR in to fix it.

Also: Just curious, is there a particular reason why `DelaunayTriangulation.jump_and_march` was insufficient for the `which_tri` function? i.e. for 

```julia
function which_tri(trivec::Vector{Vector{Vector{Float64}}}, x::Integer, y::Integer)
    findfirst(z -> 1 == inpolygon((x, y), trivec[z]), 1:length(trivec))
end
```

did `jump_and_march` have issues for you?

### PR DESCRIPTION
**Pull request by DanielVandH**
As mentioned in #12, the package currently doesn't work (it's been broken at least since #10 was merged). This PR does the following:

1. Updates the syntax of DelaunayTriangulation to 1.0 and sets the compat to be strictly `1` and above.
2. Adds tests using ReferenceTests.jl to avoid this same issue happening again.
3. Removes the dependency on PolygonOps.jl and instead uses DelaunayTriangulation.jl's `jump_and_march` for point location.
4. I also improved some of the code in other places to help with type stability. Doubt it made it much faster anyway, but just while I was noticing things.

### CODE REVIEW

File: Project.toml
Comment (jonocarroll) [inline Project.toml:1]: Would you like to be included in the `authors` field?

Comment (jonocarroll) [reply Project.toml:1]: I'm less familiar with versioning protocol in Julia - would this constitute a minor version bump?

**Comment by jonocarroll** [comment]
This is fantastic, and I'm extremely grateful for the fixes and improvements here. I'm learning a lot from what you've added/changed, so thank you.

I'm happy to include you as an author based on this PR if you would like. The version should also be updated, presumably.

Cheers!!!

File: Project.toml
Comment (DanielVandH) [reply Project.toml:1]: > Would you like to be included in the authors field?

That's alright, no need :).

> I'm less familiar with versioning protocol in Julia - would this constitute a minor version bump?

Just a minor version. I'll bump it.

**Comment by DanielVandH** [comment]
Should be good to merge now

**Comment by jonocarroll** [comment]
Many thanks!

**Pull request merged**